snap/gadget.go

```diff
@@ -27,73 +27,43 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
-type gadgetYaml struct {
-	Volumes map[string]volume `yaml:"volumes,omitempty"`
+type GadgetInfo struct {
+	Volumes map[string]GadgetVolume `yaml:"volumes,omitempty"`
+
+	// Default configuration for snaps (snap-id => key => value).
```

Comment:
snap-ids are the safest bet here, but they are going to awkward :/, we'll have to grow some tooling I suppose at some point